### ISSUE TITLE
Repeated input (self hosted)

### ISSUE THREAD

**Issue by keesj** [play-with-docker/play-with-docker#477]
Hi,

I have a self hosted version running for a demo and I noticed two problems.
When I leave the terminal for a bit longer the commands get repeated.

On the server the output is a bit hard to understand what is really going on (I am using the SDK example but I see the same on the jekyll based website) 

```
l2         | 2021/07/02 07:57:12 Saved networks
l2         | 2021/07/02 07:57:12 Saved networks
pwd        | 2021/07/02 07:57:12 Cleaned up session [c3fb3g2fe8b0i6ts0910]
pwd        | 2021/07/02 07:57:12 Unscheduled session c3fb3g2fe8b0i6ts0910
pwd        | 2021/07/02 07:57:12 EVENT: Session End c3fb3g2fe8b0i6ts0910
l2         | 2021/07/02 07:57:12 Saved networks
l2         | 2021/07/02 07:57:13 Saved networks
pwd        | [negroni] Started GET /sessions/c3fc39afe8b0i6ts0a10/ws/
pwd        | [negroni] Completed 200 OK in 50.00891768s
pwd        | 2021/07/02 07:57:24 Error reading message from websocket. Got: websocket: close 1006 (abnormal closure): unexpected EOF
pwd        | 2021/07/02 07:57:24 Session [c3fcblife8b0i6ts0a3g] doesn't have clients. Returning default viewport
pwd        | [negroni] Started GET /sessions/c3fcblife8b0i6ts0a3g/ws/
pwd        | [negroni] Completed 200 OK in 50.007812057s
pwd        | 2021/07/02 07:57:48 Error reading message from websocket. Got: websocket: close 1006 (abnormal closure): unexpected EOF
pwd        | 2021/07/02 07:57:48 Session [c3fccm2fe8b0i6ts0a50] doesn't have clients. Returning default viewport
pwd        | [negroni] Started GET /sessions/c3fccm2fe8b0i6ts0a50/ws/
pwd        | [negroni] Completed 200 OK in 50.055032315s
pwd        | 2021/07/02 07:58:17 Error reading message from websocket. Got: websocket: close 1006 (abnormal closure): unexpected EOF
pwd        | 2021/07/02 07:58:17 Session [c3fcblife8b0i6ts0a3g] doesn't have clients. Returning default viewport
pwd        | [negroni] Started GET /sessions/c3fcblife8b0i6ts0a3g/ws/
pwd        | 2021/07/02 07:58:23 Updating playgrounds configuration
pwd        | 2021/07/02 07:58:33 Error reading message from websocket. Got: websocket: close 1006 (abnormal closure): unexpected EOF
pwd        | [negroni] Completed 200 OK in 1m18.937252009s
pwd        | 2021/07/02 07:58:33 Session [c3fc39afe8b0i6ts0a10] doesn't have clients. Returning default viewport
pwd        | [negroni] Started GET /sessions/c3fc39afe8b0i6ts0a10/ws/
pwd        | 2021/07/02 07:58:40 Error reading message from websocket. Got: websocket: close 1006 (abnormal closure): unexpected EOF
pwd        | [negroni] Completed 200 OK in 50.007652776s
pwd        | 2021/07/02 07:58:40 Session [c3fccm2fe8b0i6ts0a50] doesn't have clients. Returning default viewport
pwd        | [negroni] Started GET /sessions/c3fccm2fe8b0i6ts0a50/ws/

```

**Comment by marcosnils** [comment]
thx for reporting.

> When I leave the terminal for a bit longer the commands get repeated.

When this happens, can you check if there's any activity in the network monitor under the websocket channel? Just trying to see if it's indeed a backend issue or something regarding the SDK re-drawing the screen. 

It'd be great if you can check if the same things happens in the open playground play-with-docker.com

**Comment by keesj-exset** [comment]
Hi,

I tried in firefox and chrome to monitor the websocket but I do not see any acivity even them typing(I will need to look how to do this properly) 

wss://pwd.training01.exsetlabs.com/sessions/c3fi6r9pl1i0i4p3pecg/ws/

```
pwd        | 2021/07/02 14:36:20 Error reading message from websocket. Got: websocket: close 1006 (abnormal closure): unexpected EOF
pwd        | 2021/07/02 14:36:20 Session [c3fi6r9pl1i0i4p3pecg] doesn't have clients. Returning default viewport
```

![grab](https://user-images.githubusercontent.com/62298192/124290865-3e117780-db54-11eb-90f2-0f4c49dcff44.png)

Further in the log: (after i took the screenshot)
```
pwd        | [negroni] Started GET /sessions/c3fi6r9pl1i0i4p3pecg/ws/
pwd        | [negroni] Completed 200 OK in 50.00967779s
pwd        | 2021/07/02 14:39:32 Error reading message from websocket. Got: websocket: close 1006 (abnormal closure): unexpected EOF
pwd        | 2021/07/02 14:39:32 Session [c3fhcj1pl1i0i4p3pe2g] doesn't have clients. Returning default viewport
pwd        | [negroni] Started GET /sessions/c3fhcj1pl1i0i4p3pe2g/ws/
pwd        | [negroni] Completed 200 OK in 50.006306667s

```

**Comment by marcosnils** [comment]
![image](https://user-images.githubusercontent.com/1578458/124291728-3ee1e380-db2b-11eb-9f35-dac4961c2907.png)


Select the websocket connection and check the "messages" tab in the inspector.

**Comment by keesj** [comment]
Hello,

Using the websockets and the PWD front-end I was able to find my problem. I am running a second haproxy in front of PWD and this one was not configured properly to have long running sessions 

I increased the client timeout on the haproxy front end and the server timeout on the backend and this are working much smother. 
https://gist.github.com/sourcec0de/0834e50e0470e573419f979597c701c8

After about one week of running this setup a lot of timeout sessions where kept in the sessions volume and starting up became very slow. The setup now appears more stable. I will run it for a few more days and close this issue if this indeed was the problem. Thanks for helping out!
